OPTIMIZATION: Storage, Capacity, and Prerequisites (#1118)
minio/docs: source/includes/common-installation.rst

.. end-local-jbod-single-node-desc

.. start-storage-requirements-desc

The following requirements summarize the :ref:`minio-hardware-checklist-storage` section of MinIO's hardware recommendations:

Use Local Storage
   Direct-Attached Storage (DAS) has significant performance and consistency advantages over networked storage (NAS, SAN, NFS).
   MinIO strongly recommends flash storage (NVMe, SSD) for primary or "hot" data.

Use XFS-Formatting for Drives

>    Deployments using non-XFS filesystems (EXT4, BTRFS, ZFS) tend to have lower performance and may also exhibit unexpected or undesired behavior at scale or under load.

Reviewer: Add a note about other fs being untested by us, and we use this cuz it works sort of thing

Use Consistent Type of Drive

>    Each :term:`pool` must use the same type (NVMe, SSD)  of drive with identical capacity (e.g. ``N`` TB) . 

Reviewer 2: ```suggestion
   Each :term:`pool` must use the same type (NVMe, SSD)  of drive with identical capacity (for example, all ``N`` TB) . 
```

I'd also replace N with a number, not sure what is a current typical size for our audience.

Author: Let me know how this reads.

   MinIO does not distinguish drive types and does not benefit from mixed storage types. 

Use Consistent Size of Drive
   MinIO limits the size used per drive to the smallest drive in the deployment. 
   For example, if the deployment has 15 10TB drives and 1 1TB drive, MinIO limits the per-drive capacity to 1TB.

Configure Sequential Drive Mounting

>    MinIO uses expansion notation ``{x...y}`` to denote a sequential series of drives when creating the new |deployment|, where all nodes in the |deployment| have an identical set of mounted drives. 

Reviewer: ```suggestion
   MinIO uses Go expansion notation ``{x...y}`` to denote a sequential series of drives when creating the new |deployment|, where all nodes in the |deployment| have an identical set of mounted drives. 
```

>    Configure drive mounting paths as a sequential series to best support this notation.

Reviewer 2: "to best support" doesn't feel very urgent, maybe an example would help

Persist Drive Mounting and Mapping Across Reboots

>    Use ``/etc/fstab`` or a similar file-based mount configuration to ensure consistent drive-to-mount mapping across node reboots.

Reviewer: nit : similar is confusing here since it implies there are multiple avenues in linux, maybe replace with or equivalent for non linux OS

   MinIO strongly recommends flash storage (NVMe, SSD) for primary or "hot" data.

Use XFS-Formatting for Drives
   MinIO strongly recommends provisioning XFS formatted drives for storage.
   MinIO uses XFS as part of internal testing and validation suites, providing additional confidence in performance and behavior at all scales.

   MinIO does **not** test nor recommend any other filesystem, such as EXT4, BTRFS, or ZFS.

Use Consistent Type of Drive
   MinIO does not distinguish drive types and does not benefit from mixed storage types. 
   Each :term:`pool` must use the same type (NVMe, SSD)

   For example, deploy a pool consisting of only NVMe drives.
   If you deploy some drives as SSD or HDD, MinIO treats those drives identically to the NVMe drives.
   This can result in performance issues, as some drives have differing or worse read/write characteristics and cannot respond at the same rate as the NVMe drives.

Use Consistent Size of Drive
   MinIO limits the size used per drive to the smallest drive in the deployment.

   For example, deploy a pool consisting of the same number of NVMe drives with identical capacity of ``7.68TiB``.

>    If you deploy one drive with ``3.84TiB``, MinIO treats all drives in the pool as having that capacity.

Reviewer 3: ```suggestion
   If you deploy one drive with ``3.84TiB``, MinIO treats all drives in the pool as having that smaller capacity.
```

  availability benefits when used with distributed MinIO deployments, and
  typically reduce system performance.

Use Consistent Type of Drive

>    Ensure all nodes in the |deployment| use the same type (NVMe, SSD, or HDD)  of drive with identical capacity (e.g. ``N`` TB) . 

Reviewer 2: ```suggestion
   Ensure all nodes in the |deployment| use the same type (NVMe, SSD, or HDD)  of drive with identical capacity (for example, ``N`` TB) . 
```

   MinIO does not distinguish drive types and does not benefit from mixed storage types. 

> Use Consistent Size of Drive

Reviewer: this seems like a duplicate from above? ^^

   MinIO limits the size used per drive to the smallest drive in the deployment. 
   For example, if the deployment has 15 10TB drives and 1 1TB drive, MinIO limits the per-drive capacity to 1TB.

Configure Sequential Drive Mounting

>    MinIO uses expansion notation ``{x...y}`` to denote a sequential series of drives when creating the new |deployment|, where all nodes in the |deployment| have an identical set of mounted drives. 

Reviewer: Same dupe comment
